flocker/control/_diffing.py

```diff
@@ -0,0 +1,247 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+# -*- test-case-name: flocker.control.test.test_diffing -*-
+
+from pyrsistent import (
+    PClass,
+    PMap,
+    PVector,
+    PSet,
+    field,
+    freeze,
+    pvector,
+)
+
+from zope.interface import Interface, implementer
+
+
+class _DiffChange(Interface):
+    """
+    Interface for a diff change.
+
+    This is simply something that can be applied to an object to create a new
+    object.
+
+    This interface is created more to simply documentation than for any of the
+    actual zope.interface mechanisms.
+    """
+
+    def apply(obj):
+        """
+        Apply this diff change to the passed in object and return a new object
+        that is obj with the ``self`` diff applied.
+
+        :param object obj: The object to apply the diff to.
+
+        :returns: A new object that is the passed in object with the diff
+            applied.
+        """
+        pass
+
+
+@implementer(_DiffChange)
+class _Remove(PClass):
+    """
+    A ``_DiffChange`` that removes an object from a ``PSet`` or a key from a
+    ``PMap`` inside a nested object tree.
+
+    :ivar path: The path in the nested object tree of the object to be removed
+        from the import set.
+    """
+
+    path = field(
+        type=PVector,
+        factory=freeze
+    )
```

Comment:
A little funky because I don't care what the type of the members of the field are... but that might express what this is better.